File: rust/src/models/environment_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{connector_out::ConnectorOut, event_type_out::EventTypeOut};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct EnvironmentOut {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub version: Option<i64>,

    #[serde(rename = "createdAt")]
    pub created_at: chrono::DateTime<chrono::Utc>,

    #[serde(rename = "eventTypes")]
    pub event_types: Vec<EventTypeOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub settings: Option<serde_json::Value>,

    pub connectors: Vec<ConnectorOut>,
}

impl EnvironmentOut {
    pub fn new(
        created_at: chrono::DateTime<chrono::Utc>,
        event_types: Vec<EventTypeOut>,
        connectors: Vec<ConnectorOut>,
    ) -> Self {
        Self {
            version: None,
            created_at,
            event_types,
            settings: None,
            connectors,
        }
    }
}

File: javascript/src/models/environmentOut.ts
// this file is @generated
import { type ConnectorOut, ConnectorOutSerializer } from "./connectorOut";
import { type EventTypeOut, EventTypeOutSerializer } from "./eventTypeOut";

export interface EnvironmentOut {
  version?: number;
  createdAt: Date;
  eventTypes: EventTypeOut[];
  settings: any | null;
  connectors: ConnectorOut[];
}

export const EnvironmentOutSerializer = {
  _fromJsonObject(object: any): EnvironmentOut {
    return {
      version: object["version"],
      createdAt: new Date(object["createdAt"]),
      eventTypes: object["eventTypes"].map((item: EventTypeOut) =>
        EventTypeOutSerializer._fromJsonObject(item)
      ),
      settings: object["settings"],
      connectors: object["connectors"].map((item: ConnectorOut) =>
        ConnectorOutSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: EnvironmentOut): any {
    return {
      version: self.version,
      createdAt: self.createdAt,
      eventTypes: self.eventTypes.map((item) =>
        EventTypeOutSerializer._toJsonObject(item)
      ),
      settings: self.settings,
      connectors: self.connectors.map((item) =>
        ConnectorOutSerializer._toJsonObject(item)
      ),
    };
  },
};
